### PR TITLE
feat(TMC-15843): disabled item in datalist

### PR DESCRIPTION
**Pull request by VolodymyrKovalM**
**What is the problem this PR is trying to solve?**
https://jira.talendforge.org/browse/TMC-15843
In TMC we need a possibility to make disabled item in Datalist

**What is the chosen solution to this problem?**

**Please check if the PR fulfills these requirements**

* [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
* [x] Tests for the changes have been added (for bug fixes / features) And [non reg](./screenshots.md) done before need review
* [ ] Docs have been added / updated (for bug fixes / features)
* [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

### CODE REVIEW

File: packages/components/src/Datalist/Datalist.component.js [hunk 1]
```diff
@@ -170,6 +170,10 @@ class Datalist extends Component {
 		if (this.props.multiSection) {
 			newValue = this.state.suggestions[sectionIndex].suggestions[itemIndex];
 		}
+		if (newValue.disabled) {
```
Comment (frassinier) [inline packages/components/src/Datalist/Datalist.component.js]: We need to follow that PR I guess https://github.com/moroshko/react-autowhatever/pull/35

Comment (VolodymyrKovalM) [reply packages/components/src/Datalist/Datalist.component.js]: Yes, also saw that PR, but it was done around 4 months ago (
But we can follow of course

Comment (VolodymyrKovalM) [reply packages/components/src/Datalist/Datalist.component.js]: Yes, also saw that PR, but it was done around 4 months ago (
But we can follow of course

**Comment by frassinier** [comment]
I validated even if I consider that disabled items should not be reachable by keyboard like a simple `<select>` does actually.

**Pull request merged**